src/main/java/cs2103/v15_1j/jimjim/JJMain.java

```diff
@@ -0,0 +1,46 @@
+package cs2103.v15_1j.jimjim;
+
+import javafx.application.Application;
+import javafx.stage.Stage;
+
+public class JJMain extends Application {
+
+	private Controller con;
+	private Storage storage;
+	private Parser parser;
+	private UI ui;
+	private final String TASK_FILE_NAME = "tasks.json";
+	private final String EVENT_FILE_NAME = "events.json";
+
+	public JJMain(){
+		con = new JJController();
+		storage = new JJStorage();
+		parser = new JJParser();
+		ui = new JJUI();
+
+		storage.setSaveFiles(TASK_FILE_NAME, EVENT_FILE_NAME);
+		con.setParser(parser);
+		con.setStorage(storage);
+	}
+
+	public JJMain(Controller con, Storage storage, Parser parser, UI ui){
```

Comment:
I thought this class is the entry point, so no other class will create an instance of this class? If that's the case, why do we need this constructor?